print-ospf6.c

```diff
@@ -388,14 +388,23 @@ static int
 ospf6_print_lshdr(netdissect_options *ndo,
                   const struct lsa6_hdr *lshp, const u_char *dataend)
 {
+	u_int ls_length;
+
 	if ((const u_char *)(lshp + 1) > dataend)
 		goto trunc;
 
+	ls_length = GET_BE_U_2(lshp->ls_length);
+	if (ls_length < sizeof(struct lsa_hdr)) {
+		ND_PRINT("\n\t	  Bogus length %u < header (%zu)", ls_length,
+		    sizeof(struct lsa_hdr));
+		goto trunc;
```

Comment:
It's perhaps more an "invalid" length case than a "truncated" case ?

Reply:
> It's perhaps more an "invalid" length case than a "truncated" case ?

My thinking is "this TLV doesn't have enough data to contain the header of the TLV, so, the TLV itself is truncated".

All 3 callers of `ospf6_print_lshdr()` use `if (ospf6_print_lshdr(...)) return(1);`.  From a skim, I don't see a reason for any of them to distinguish between "truncated" and "invalid in this specific way".

Reply:
The notion of truncation may have evolved over the years, but currently a truncation occurs when we try to get len bytes in the packet buffer and we can't.
(checks via GET_\*, ND_TTEST_\*, ND_TCHECK_\*, etc. macros). For ND_TCHECK_\* macros, if ND_LONGJMP_FROM_TCHECK is defined for a printer, a `longjmp` occurs, no more `goto trunc` (Printers "Modernized" for the 5.0.0 release, remain 34 to do.)
The macros for testing invalid length cases are the ND_ICHECK*, with a `goto invalid`.
If we add in line 396 the following command, we get the value 470, with the test file `ospf-signed-integer-ubsan.pcap`:
`ND_PRINT("[[%u]]", ND_BYTES_AVAILABLE_AFTER(lshp->ls_length));`
Thus there is no truncation here.
Tshark says also the packet is `malformed` (like our `invalid`.), not `[Packet size limited during capture: XXX truncated]`:
```
[...]
            Link State ID: 145.245.255.254
            Advertising Router: 204.122.189.255
            Sequence Number: 0x02000000
            Checksum: 0x0000
            Length: 0
            [Expert Info (Warning/Protocol): Unknown LSA Type 224]
                [Unknown LSA Type 224]
                [Severity level: Warning]
                [Group: Protocol]
[Malformed Packet: OSPF]
    [Expert Info (Error/Malformed): Malformed Packet (Exception occurred)]
        [Malformed Packet (Exception occurred)]
        [Severity level: Error]
        [Group: Malformed]
```
Perhaps this change should be updated after a modernization of print-ospf6.c.

Reply:
Perhaps. I'm not prepared to make a major rework as a prerequisite for my contribution, so if it's better to have undefined behavior than to print an incorrect truncation message, great, feel free to discard my changes.

Reply:
If the proposed change makes the code safer, perhaps we could address any cosmetic issues after merging.

Reply:
My first idea was to hold the change waiting the modernization of print-ospf6.c, but as Denis says we can do the fix as a temporary change. I will commit this part. Thanks for your work.

Reply:
> My first idea was to hold the change waiting the modernization of print-ospf6.c, but as Denis says we can do the fix as a temporary change. I will commit this part. Thanks for your work.

Thank you.  Undefined behavior does not necessarily lead to incorrect behavior, but, if you enable undefined behavior detection in your fuzzing environment, having undefined behavior can cause so much noise that it can prevent you from discovering real problems via fuzzing.